dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -762,6 +780,39 @@ public Limits setUserLimits(@ApiParam(hidden = true) @Auth User authUser,
         return getStrippedWorkflowsAndServices(userDAO.findById(user.getId()));
     }
 
+    @POST
+    @Path("/workflow")
+    @Timed
+    @UnitOfWork
+    @Operation(operationId = "addUserToDockstoreWorkflows", description = "Adds a user to any Dockstore workflows that they should have access to.", security = @SecurityRequirement(name = ResourceConstants.OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    @ApiOperation(value = "Adds a user to any Dockstore workflows that they should have access to.", authorizations = {
+            @Authorization(value = JWT_SECURITY_DEFINITION_NAME) },
+            response = Workflow.class, responseContainer = "List")
+    public List<Workflow> addUserToDockstoreWorkflows(@ApiParam(hidden = true) @Parameter(hidden = true, name = "user", in = ParameterIn.HEADER) @Auth User authUser) {
+        final User user = userDAO.findById(authUser.getId());
+        // Ignore hosted workflows
+        List<SourceControl> sourceControls = Arrays.stream(SourceControl.values()).filter(sourceControl -> !Objects.equals(sourceControl, SourceControl.DOCKSTORE)).collect(
+                Collectors.toList());
+
+        List<Token> scTokens = getAndRefreshTokens(user, tokenDAO, client, bitbucketClientID, bitbucketClientSecret)
+                .stream()
+                .filter(token -> sourceControls.contains(token.getTokenSource().getSourceControl()))
+                .collect(Collectors.toList());
+
+        scTokens.stream().forEach(token -> {
+            SourceCodeRepoInterface sourceCodeRepo =  SourceCodeRepoFactory.createSourceCodeRepo(token, client);
+            Map<String, String> gitUrlToRepositoryId = sourceCodeRepo.getWorkflowGitUrl2RepositoryId();
+            Set<String> organizations = gitUrlToRepositoryId.values().stream().map(repository -> repository.split("/")[0]).collect(Collectors.toSet());
+
+            organizations.forEach(organization -> {
+                List<Workflow> workflows = workflowDAO.findByOrganization(token.getTokenSource().getSourceControl(), organization);
+                workflows.stream().forEach(workflow -> workflow.getUsers().add(user));
+            });
+        });
+
+        return getStrippedBioworkflows(userDAO.findById(user.getId()));
```

Comment:
Fine for now, someone should probably just call similar code to the userWorkflows endpoint after it's been optimized